src/solver/optimisation/opt_rename_problem.cpp

```diff
@@ -0,0 +1,396 @@
+                  LocationIdentifier(currentAssetsStorage.area, ExportStructLocationDict::area),
+                  TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::hour));
+}
+
+void ConstraintNamer::MinHydroPower()
+{
+    currentAssetsStorage.problem
+      ->NomDesContraintes[currentAssetsStorage.problem->NombreDeContraintes - 1]
+      = BuildName(toString(ExportStructConstraintsDict::MinHydroPower),
+                  LocationIdentifier(currentAssetsStorage.area, ExportStructLocationDict::area),
+                  TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::week));
+}
+
+void ConstraintNamer::MaxHydroPower()
+{
+    currentAssetsStorage.problem
+      ->NomDesContraintes[currentAssetsStorage.problem->NombreDeContraintes - 1]
+      = BuildName(toString(ExportStructConstraintsDict::MaxHydroPower),
+                  LocationIdentifier(currentAssetsStorage.area, ExportStructLocationDict::area),
+                  TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::week));
+}
+
+void ConstraintNamer::MaxPumping()
+{
+    currentAssetsStorage.problem
+      ->NomDesContraintes[currentAssetsStorage.problem->NombreDeContraintes - 1]
+      = BuildName(toString(ExportStructConstraintsDict::MaxPumping),
+                  LocationIdentifier(currentAssetsStorage.area, ExportStructLocationDict::area),
+                  TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::week));
+}
+
+void ConstraintNamer::AreaHydroLevel()
+{
+    currentAssetsStorage.problem
+      ->NomDesContraintes[currentAssetsStorage.problem->NombreDeContraintes - 1]
+      = BuildName(toString(ExportStructConstraintsDict::AreaHydroLevel),
+                  LocationIdentifier(currentAssetsStorage.area, ExportStructLocationDict::area),
+                  TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::hour));
+}
+
+void ConstraintNamer::FinalStockEquivalent()
+{
+    currentAssetsStorage.problem
+      ->NomDesContraintes[currentAssetsStorage.problem->NombreDeContraintes - 1]
+      = BuildName(toString(ExportStructConstraintsDict::FinalStockEquivalent),
+                  LocationIdentifier(currentAssetsStorage.area, ExportStructLocationDict::area),
+                  TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::hour));
+}
+
+void ConstraintNamer::FinalStockExpression()
+{
+    currentAssetsStorage.problem
+      ->NomDesContraintes[currentAssetsStorage.problem->NombreDeContraintes - 1]
+      = BuildName(toString(ExportStructConstraintsDict::FinalStockExpression),
+                  LocationIdentifier(currentAssetsStorage.area, ExportStructLocationDict::area),
+                  TimeIdentifier(currentAssetsStorage.timeStep, ExportStructTimeStepDict::hour));
+}
+
+void ConstraintNamer::BindingConstraint(const std::string& name, ExportStructTimeStepDict type)
```

Comment:
```C++
auto timeStepType
      = (type == ExportStructTimeStepDict::hour)  ? ExportStructBindingConstraintType::hourly
        : (type == ExportStructTimeStepDict::day) ? ExportStructBindingConstraintType::daily
                                                  : ExportStructBindingConstraintType::weekly;
```
Why is it so complicated ?
(Without use of **ExportStructTimeStepDict**, it would be much more simple by the way)